tests/contrib/compare_ids_test_fail_test.sh.tpl

```diff
@@ -0,0 +1,46 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The test script generated by the create_failing_test_for_compare_ids_test test rule
+# This creates a new workspace and copies all required files into it, then runs the test
+# which is expected to fail, and passes only if it fails.
+
+#!/bin/bash
+
+set -eux
+
+mkdir -p new_temp_workspace_{name}
+cd new_temp_workspace_{name}
+
+touch WORKSPACE
+
+echo > BUILD {test_code}
+
+ln ../{bzl_path}
+
+ln ../{tpl_path}
+
+k=0
+for i in {tars}
+do
+  mv $(ln -v ../$i | cut -d "'" -f2) $k.tar
+  k=`expr $k + 1`
```

Comment:
In contrib/extract_image_id.sh, you use i=$(expr $i + 1) format. Please be consistent. Also, if you can give the indices more meaningful names, that will be better.